standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestGetAllTableConstraints.java

```diff
@@ -0,0 +1,382 @@
+    Assert.assertTrue(fetched.getForeignKeys().isEmpty());
+
+  }
+
+  @Test
+  public void allConstraintsPresent() throws TException {
+    Table table = testTables[0];
+    Table parentTable = testTables[1];
+
+    // Set col1 as primary key Constraint in default catalog and database
+    String pkConstraintName = "col1_pk";
+    List<SQLPrimaryKey> pk =
+        new SQLPrimaryKeyBuilder().onTable(table).addColumn("col1").setConstraintName(pkConstraintName)
+            .build(metaStore.getConf());
+    client.addPrimaryKey(pk);
+
+    // Set col2 with Unique Constraint in default catalog and database
+    String uniqueConstraintName = "col2_unique";
+    List<SQLUniqueConstraint> uc =
+        new SQLUniqueConstraintBuilder().onTable(table).addColumn("col2").setConstraintName(uniqueConstraintName)
+            .build(metaStore.getConf());
+    client.addUniqueConstraint(uc);
+
+    // Set col3 with default Constraint in default catalog and database
+    String defaultConstraintName = "col3_default";
+    List<SQLDefaultConstraint> dv =
+        new SQLDefaultConstraintBuilder().onTable(table).addColumn("col3").setConstraintName(defaultConstraintName)
+            .setDefaultVal(false).build(metaStore.getConf());
+    client.addDefaultConstraint(dv);
+
+    // Set col3 with not null constraint in default catalog and database;
+    String nnCol3ConstraintName = "col3_not_null";
+    List<SQLNotNullConstraint> nnCol2 =
+        new SQLNotNullConstraintBuilder().onTable(table).addColumn("col3").setConstraintName(nnCol3ConstraintName)
+            .build(metaStore.getConf());
+    client.addNotNullConstraint(nnCol2);
+
+    // Set col2 with not check constraint in default catalog and database;
+    String ccCol2ConstraintName = "col2_check";
+    List<SQLCheckConstraint> cc =
+        new SQLCheckConstraintBuilder().onTable(table).addColumn("col2").setConstraintName(ccCol2ConstraintName)
+            .setCheckExpression("= 5").build(metaStore.getConf());
+    client.addCheckConstraint(cc);
+
+    // Set col1 of parent table to PK and Set Col4 of table to FK
+    String parentPkConstraintName = "parentpk";
+    List<SQLPrimaryKey> parentPk =
+        new SQLPrimaryKeyBuilder().onTable(parentTable).addColumn("col1").setConstraintName(parentPkConstraintName)
+            .build(metaStore.getConf());
+    client.addPrimaryKey(parentPk);
+    String fkConstraintName = "fk";
+    List<SQLForeignKey> fk =
+        new SQLForeignKeyBuilder().fromPrimaryKey(parentPk).onTable(table).setConstraintName(fkConstraintName)
+            .addColumn("col4").build(metaStore.getConf());
+    client.addForeignKey(fk);
+
+    // Fetch all constraints for the table in default catalog and database
+    AllTableConstraintsRequest request = new AllTableConstraintsRequest(table.getDbName(), table.getTableName());
+    request.setCatName(table.getCatName());
+    SQLAllTableConstraints fetched = client.getAllTableConstraints(request);
+
+    // Assert primary key constraint
```

Comment:
Too much of duplicate code. Can write common methods for the asserts and use it in all tests.